lib/arel/visitors/to_sql.rb

```diff
@@ -166,6 +166,28 @@ def visit_Arel_Nodes_False o, collector
         collector << "FALSE"
       end
 
+      def visit_Arel_Nodes_ValuesList o, collector
+        collector << "VALUES "
+
+        len = o.rows.length - 1
+        o.rows.each_with_index { |row, i|
+          collector << '('
+          row_len = row.length - 1
+          row.each_with_index do |value, k|
```

Comment:
@matthewd now when we iterate on each element of the row the method gets a bit complex. How do you feel about it?

Ideally `Arel::ValuesList` would be a container for `Arel::Values`. Unfortunately we can't just do it because `visit_Arel_Nodes_Values` appends `VALUES` on every call which is something that we don't want for `ValuesList` entries.

I tried to implement `Values` visitor in a way that it wouldn't always append `VALUES` but that broke a bunch of other use cases.